dali/imgcodec/parsers/webp.cc

```diff
@@ -102,65 +133,122 @@ std::string sequence_of_integers(const std::array<uint8_t, N> &data) {
   return result;
 }
 
+// If chunk_size is odd, there is a one byte padding that must be skipped
+void skip_rest_of_chunk(InputStream &stream,
+                        const ChunkHeader &header,
+                        size_t bytes_of_data_read = 0) {
+  stream.Skip(header.chunk_size + header.chunk_size % 2 - bytes_of_data_read);
+}
+
+void fetch_info_from_exif_data(std::vector<uint8_t> &data, ImageInfo &info) {
+  cv::ExifReader reader;
+  reader.parseExif(data.data(), data.size());
+  const auto orientation_value = reader.getTag(cv::ORIENTATION).field_u16;
+  const auto exif_orientation = static_cast<ExifOrientation>(orientation_value);
+  info.orientation = FromExifOrientation(exif_orientation);
+}
+
+// Iterate over the chunks seeking the EXIF chunk
+// InputStream must be at the start of a chunk
+void seek_exif_data(InputStream &stream, ImageInfo &info) {
+  while (true) {
+    const auto header = stream.ReadOne<ChunkHeader>();
+    if (header.identifier == tag("EXIF")) {
+      // Parse the chunk data into the orientation
+      std::vector<uint8_t> buffer(header.chunk_size);
+      stream.ReadBytes(buffer.data(), buffer.size());
+      fetch_info_from_exif_data(buffer, info);
+      break;
+    } else {
+      // Skip the rest of the chunk
+      skip_rest_of_chunk(stream, header);
+    }
+  }
+}
+
 }  // namespace
 
 ImageInfo WebpParser::GetInfo(ImageSource *encoded) const {
   auto stream = encoded->Open();
   ImageInfo info;
 
   stream->Skip<RiffHeader>();
-  const auto vp8_header = stream->ReadOne<vp8_header_t>();
+  auto vp8_header = stream->ReadOne<ChunkHeader>();
+  to_little_endian(vp8_header);
+
   if (is_simple_lossy_format(vp8_header)) {
-    const auto lossy_header = stream->ReadOne<WebpLossyHeader>();
+    auto lossy_header = stream->ReadOne<WebpLossyHeader>();
+    to_little_endian(lossy_header);
+
     if (!is_sync_code_valid(lossy_header)) {
       DALI_FAIL("Sync code 157 1 42 not found at expected position. Found " +
                 sequence_of_integers(lossy_header.sync_code) + " instead");
     }
 
-    const int w = ReadValueLE<uint16_t>(*stream) & 0x3FFF;
-    const int h = ReadValueLE<uint16_t>(*stream) & 0x3FFF;
+    // only the last 14 bits of the fields code the dimensions
+    const int w = lossy_header.width_field & 0x3FFF;
```

Comment:
Please name this MASK; `Could be unsigned((1<<14)-1)` then i'll trust you it is `0x3fff` :)